File: app/src/main/java/com/example/breezil/favnews/ui/adapter/DiffUtils.java
package com.example.breezil.favnews.ui.adapter;

import android.support.annotation.Nullable;
import android.support.v7.util.DiffUtil;

import com.example.breezil.favnews.model.Articles;

import java.util.List;

public class DiffUtils extends DiffUtil.Callback {

    private List<Articles> oldArticleList;
    private List<Articles> newArticleList;

    public DiffUtils(List<Articles> oldArticleList, List<Articles> newArticleList) {
        this.oldArticleList = oldArticleList;
        this.newArticleList = newArticleList;
    }

    @Override
    public int getOldListSize() {
        return oldArticleList.size();
    }

    @Override
    public int getNewListSize() {
        return newArticleList.size();
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        return oldArticleList.get(oldItemPosition).getTitle() == newArticleList.get(newItemPosition).getTitle();
    }


    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        Articles oldArticle = oldArticleList.get(oldItemPosition);
        Articles newArticle = newArticleList.get(newItemPosition);

        return oldArticle.getTitle().equals(newArticle.getTitle());
    }

    @Nullable
    @Override
    public Object getChangePayload(int oldItemPosition, int newItemPosition) {
        return super.getChangePayload(oldItemPosition, newItemPosition);
    }
}

File: app/src/main/java/com/fave/breezil/fave/ui/adapter/DiffUtils.kt
package com.fave.breezil.fave.ui.adapter

import android.support.v7.util.DiffUtil

import com.fave.breezil.fave.model.Articles

class DiffUtils(private val oldArticleList: List<Articles>, private val newArticleList: List<Articles>) : DiffUtil.Callback() {

    override fun getOldListSize(): Int {
        return oldArticleList.size
    }

    override fun getNewListSize(): Int {
        return newArticleList.size
    }

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return oldArticleList[oldItemPosition].title === newArticleList[newItemPosition].title
    }


    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        val oldArticle = oldArticleList[oldItemPosition]
        val newArticle = newArticleList[newItemPosition]

        return oldArticle.title == newArticle.title
    }

    override fun getChangePayload(oldItemPosition: Int, newItemPosition: Int): Any? {
        return super.getChangePayload(oldItemPosition, newItemPosition)
    }
}
